# Homework 02

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("videosUS.csv")

In [15]:
# How the data looks like
df.head(3)

video_id trending_date  \
0  2kyS6SvSYSE      17.14.11   
1  1ZAPwfrtAFY      17.14.11   
2  5qpjK5DgCt4      17.14.11   

                                               title    channel_title  \
0                 WE WANT TO TALK ABOUT OUR MARRIAGE     CaseyNeistat   
1  The Trump Presidency: Last Week Tonight with J...  LastWeekTonight   
2  Racist Superman | Rudy Mancuso, King Bach & Le...     Rudy Mancuso   

   category_id              publish_time  \
0         22.0  2017-11-13T17:13:01.000Z   
1         24.0  2017-11-13T07:30:00.000Z   
2         23.0  2017-11-12T19:05:24.000Z   

                                                tags      views     likes  \
0                                    SHANtell martin   748374.0   57527.0   
1  last week tonight trump presidency|"last week ...  2418783.0   97185.0   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...  3191434.0  146033.0   

   dislikes  comment_count                                  thumbnail_link  \
0    2966.0        15954.0  https://i.ytimg.com/vi/2kyS6SvSYSE/default.jpg   
1    6146.0        12703.0  https://i.ytimg.com/vi/1ZAPwfrtAFY/default.jpg   
2    5339.0         8181.0  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   

  comments_disabled ratings_disabled video_error_or_removed  \
0             False            False                  False   
1             False            False                  False   
2             False            False                  False   

                                         description  
0  SHANTELL'S CHANNEL - https://www.youtube.com/s...  
1  One year after the presidential election, John...  
2  WATCH MY PREVIOUS VIDEO ▶ \n\nSUBSCRIBE ► http...

In [6]:
len(df)

41415

### How many missing values within the dataset 

#### In each column

In [12]:
df.isna().sum()

video_id                     0
trending_date              466
title                      466
channel_title              466
category_id                466
publish_time               466
tags                       466
views                      466
likes                      466
dislikes                   466
comment_count              466
thumbnail_link             466
comments_disabled          466
ratings_disabled           466
video_error_or_removed     466
description               1036
dtype: int64

In [14]:
df[df.title.isna()]

video_id trending_date title  \
53     \nSubscribe to WWE on YouTube: http://bit.ly/1...           NaN   NaN   
54                 \nVisit WWE.com: http://goo.gl/akf0J4           NaN   NaN   
55     \nMust-See WWE videos on YouTube: https://goo....           NaN   NaN   
319    \nSubscribe to WWE on YouTube: http://bit.ly/1...           NaN   NaN   
320                \nVisit WWE.com: http://goo.gl/akf0J4           NaN   NaN   
...                                                  ...           ...   ...   
36570  Follow me on Twitter ► https://twitter.com/mar...           NaN   NaN   
36571  Like me on Facebook ► https://www.facebook.com...           NaN   NaN   
36572  Join us on Reddit! ► https://www.reddit.com/r/...           NaN   NaN   
36573  Horror Outro ► https://soundcloud.com/shurkoff...           NaN   NaN   
36574  Happy Outro ► https://soundcloud.com/hielia/mi...           NaN   NaN   

      channel_title  category_id publish_time tags  views  likes  dislikes  \
53              NaN          NaN          NaN  NaN    NaN    NaN       NaN   
54              NaN          NaN          NaN  NaN    NaN    NaN       NaN   
55              NaN          NaN          NaN  NaN    NaN    NaN       NaN   
319             NaN          NaN          NaN  NaN    NaN    NaN       NaN   
320             NaN          NaN          NaN  NaN    NaN    NaN       NaN   
...             ...          ...          ...  ...    ...    ...       ...   
36570           NaN          NaN          NaN  NaN    NaN    NaN       NaN   
36571           NaN          NaN          NaN  NaN    NaN    NaN       NaN   
36572           NaN          NaN          NaN  NaN    NaN    NaN       NaN   
36573           NaN          NaN          NaN  NaN    NaN    NaN       NaN   
36574           NaN          NaN          NaN  NaN    NaN    NaN       NaN   

       comment_count thumbnail_link comments_disabled ratings_disabled  \
53               NaN            NaN               NaN              NaN   
54               NaN            NaN               NaN              NaN   
55               NaN            NaN               NaN              NaN   
319              NaN            NaN               NaN              NaN   
320              NaN            NaN               NaN              NaN   
...              ...            ...               ...              ...   
36570            NaN            NaN               NaN              NaN   
36571            NaN            NaN               NaN              NaN   
36572            NaN            NaN               NaN              NaN   
36573            NaN            NaN               NaN              NaN   
36574            NaN            NaN               NaN              NaN   

      video_error_or_removed description  
53                       NaN         NaN  
54                       NaN         NaN  
55                       NaN         NaN  
319                      NaN         NaN  
320                      NaN         NaN  
...                      ...         ...  
36570                    NaN         NaN  
36571                    NaN         NaN  
36572                    NaN         NaN  
36573                    NaN         NaN  
36574                    NaN         NaN  

[466 rows x 16 columns]

#### My assumption about why we have these missing values

Looking at the previous two results, we can clearly see that there's 466 missing value in each column (except the description, and video_id columns), and actually, these 466 missing values are common to the same videos. Maybe it means that these videos were deleted or can't be shared on public. Therefore, we can delete them.
The description column though, has 1036 missing values, which is pretty normal I guess, since descriptions are usually optional.

#### Getting rid of the 466 columns

In [16]:
df.drop(df[(df.title.isna())].index, inplace=True)

In [18]:
df[df.title.isna()]

Empty DataFrame
Columns: [video_id, trending_date, title, channel_title, category_id, publish_time, tags, views, likes, dislikes, comment_count, thumbnail_link, comments_disabled, ratings_disabled, video_error_or_removed, description]
Index: []

### Mean , Median , and Quartiles of each numerical variable within this dataset.

In [20]:
columns = ["views", "likes", "dislikes", "comment_count"]

#### Mean

In [23]:
for column in columns:
    print("Mean of", column, ":", df[column].mean())

Mean of views : 2360784.6382573447
Mean of likes : 74266.7024347359
Mean of dislikes : 3711.400888910596
Mean of comment_count : 8446.803682629612


#### Median

In [24]:
for column in columns:
    print("Median of", column, ":", df[column].median())

Median of views : 681861.0
Median of likes : 18091.0
Median of dislikes : 631.0
Median of comment_count : 1856.0


#### Quartiles

In [29]:
for column in columns:
    print("Quartiles of", column)
    print(" 25% :", df[column].quantile(0.25))
    print(" 50% :", df[column].quantile(0.50))
    print(" 75% :", df[column].quantile(0.75))
    print("\n")

Quartiles of views
 25% : 242329.0
 50% : 681861.0
 75% : 1823157.0


Quartiles of likes
 25% : 5424.0
 50% : 18091.0
 75% : 55417.0


Quartiles of dislikes
 25% : 202.0
 50% : 631.0
 75% : 1938.0


Quartiles of comment_count
 25% : 614.0
 50% : 1856.0
 75% : 5755.0




I could have used df.describe() but I just wanted to display only Mean, Mediane, and Quartiles, and I didn't want to take the category_id into consideration, I don't know if it's good or not.

### Spot the possible outliers in your dataset and get rid of them.

#### Calculating the z-scores

In [30]:
df["views_zscore"] = (df.views - df.views.mean()) / df.views.std()
df["likes_zscore"] = (df.likes - df.likes.mean()) / df.likes.std()
df["dislikes_zscore"] = (df.dislikes - df.dislikes.mean()) / df.dislikes.std()
df["comment_count_zscore"] = (df.comment_count - df.comment_count.mean()) / df.comment_count.std()